### PR TITLE
[WFCORE-7079] Remove ModuleIdentifier usage from Version module

### PR DESCRIPTION
**Pull request by yersan**
Jira issue: https://issues.redhat.com/browse/WFCORE-7079

### CODE REVIEW

File: version/src/main/java/org/jboss/as/version/ProductConfig.java [hunk 1]
```diff
@@ -262,10 +261,10 @@ private static void safeClose(Closeable c) {
 
     private static class ProductConfProps {
         private final Properties miscProperties;
-        private final ModuleIdentifier productModuleId;
+        private final String productModuleId;
 
         private ProductConfProps(String slot) {
-            this.productModuleId = slot == null ? null : ModuleIdentifier.create("org.jboss.as.product", slot);
+            this.productModuleId = slot == null ? null : "org.jboss.as.product:" + slot;
```
Comment (bstansberry) [inline version/src/main/java/org/jboss/as/version/ProductConfig.java:267]: @yersan This needs to be:

this.productModuleId = slot == null ? null : ("main",equals(slot) ? "org.jboss.as.product" : "org.jboss.as.product:" + slot);

The canonical string representation of a module with slot 'main' is to NOT include ":main".  The ModuleIdentifier class implements that logic. Where we discard its use we need to implement it ourselves.

I want to add utility methods that we can call throughout WF to handle this kind of thing, somewhere in the 'controller' or 'server' modules, but for the 'version' module I think I just needs to be manually coded.

**Review by bstansberry — CHANGES_REQUESTED** [review]

File: version/src/main/java/org/jboss/as/version/ProductConfig.java (hunk 1 above)
Comment (bstansberry) [reply version/src/main/java/org/jboss/as/version/ProductConfig.java:267]: #6270is the proposed utility. But, not usable here. Not really needed either, as the use case here is simple. That PR has a lot of code related to escaping arbitrary inputs but I think this code can assume that any slot value is simple text that doesn't need escaping.

Comment (yersan) [reply version/src/main/java/org/jboss/as/version/ProductConfig.java]: @bstansberry Done

**Review by bstansberry — APPROVED** [review]

**Comment by bstansberry** [comment]
Thanks @yersan

**Pull request merged**